homeassistant/components/google_cloud/tts.py

```diff
@@ -0,0 +1,236 @@
+"""Support for the Google Cloud TTS service."""
+import logging
+import os
+import re
+import asyncio
+import async_timeout
+import voluptuous as vol
+from google.cloud import texttospeech
+from homeassistant.components.tts import CONF_LANG, PLATFORM_SCHEMA, Provider
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_KEY_FILE = 'key_file'
+CONF_GENDER = 'gender'
+CONF_VOICE = 'voice'
+CONF_ENCODING = 'encoding'
+CONF_SPEED = 'speed'
+CONF_PITCH = 'pitch'
+CONF_GAIN = 'gain'
+CONF_PROFILES = 'profiles'
+
+SUPPORTED_LANGUAGES = [
+    'da-DK', 'de-DE', 'en-AU', 'en-GB', 'en-US', 'es-ES', 'fr-CA', 'fr-FR',
+    'it-IT', 'ja-JP', 'ko-KR', 'nb-NO', 'nl-NL', 'pl-PL', 'pt-BR', 'pt-PT',
+    'ru-RU', 'sk-SK', 'sv-SE', 'tr-TR', 'uk-UA',
+]
+DEFAULT_LANG = 'en-US'
+
+DEFAULT_GENDER = 'NEUTRAL'
+
+VOICE_REGEX = r'[a-z]{2}-[A-Z]{2}-(Standard|Wavenet)-[A-Z]|'
+DEFAULT_VOICE = ''
+
+DEFAULT_ENCODING = 'OGG_OPUS'
+
+MIN_SPEED = 0.25
+MAX_SPEED = 4.0
+DEFAULT_SPEED = 1.0
+
+MIN_PITCH = -20.0
+MAX_PITCH = 20.0
+DEFAULT_PITCH = 0
+
+MIN_GAIN = -96.0
+MAX_GAIN = 16.0
+DEFAULT_GAIN = 0
+
+SUPPORTED_PROFILES = [
+    "wearable-class-device",
+    "handset-class-device",
+    "headphone-class-device",
+    "small-bluetooth-speaker-class-device",
+    "medium-bluetooth-speaker-class-device",
+    "large-home-entertainment-class-device",
+    "large-automotive-class-device",
+    "telephony-class-application",
+]
+
+SUPPORTED_OPTIONS = [
+    CONF_VOICE,
+    CONF_GENDER,
+    CONF_ENCODING,
+    CONF_SPEED,
+    CONF_PITCH,
+    CONF_GAIN,
+    CONF_PROFILES,
+]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_KEY_FILE): cv.string,
+    vol.Optional(CONF_LANG, default=DEFAULT_LANG):
+        vol.In(SUPPORTED_LANGUAGES),
+    vol.Optional(CONF_GENDER, default=DEFAULT_GENDER):
+        vol.All(vol.Upper, vol.In(
+            texttospeech.enums.SsmlVoiceGender.__members__
+        )),
+    vol.Optional(CONF_VOICE, default=DEFAULT_VOICE):
+        cv.matches_regex(VOICE_REGEX),
+    vol.Optional(CONF_ENCODING, default=DEFAULT_ENCODING):
+        vol.All(vol.Upper, vol.In(
+            texttospeech.enums.AudioEncoding.__members__
+        )),
+    vol.Optional(CONF_SPEED, default=DEFAULT_SPEED):
+        vol.All(vol.Coerce(float), vol.Clamp(min=MIN_SPEED, max=MAX_SPEED)),
+    vol.Optional(CONF_PITCH, default=DEFAULT_PITCH):
+        vol.All(vol.Coerce(float), vol.Clamp(min=MIN_PITCH, max=MAX_PITCH)),
+    vol.Optional(CONF_GAIN, default=DEFAULT_GAIN):
+        vol.All(vol.Coerce(float), vol.Clamp(min=MIN_GAIN, max=MAX_GAIN)),
+    vol.Optional(CONF_PROFILES, default=[]):
+        vol.All(cv.ensure_list, [vol.In(SUPPORTED_PROFILES)]),
+})
+
+
+async def async_get_engine(hass, config):
+    """Set up Google Cloud TTS component."""
+    key_file = config.get(CONF_KEY_FILE)
+    if key_file:
+        key_file = hass.config.path(key_file)
```

Comment:
Can you please explain than how user would reference that? Is there any way to reference `config` folder in `configuration.yaml`? Something like:
```yaml
tts:
  - platform: google_cloud
    key_file: "<config>/myfile.json"
```
 Or is user even able to put file elsewhere?
I'm using **Hass.io** that's why I'm asking. Because I only have access to `config` folder.

Reply:
I suggest we keep `hass.config.path`. It means that if we specify a relative path, it is resolved relatively to the config dir. If it's an absolute path, it remains absolute.